replication-api-impl/src/main/java/com/connexta/replication/api/impl/queue/memory/MemoryQueueBroker.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.api.impl.queue.memory;
+
+import com.connexta.replication.api.queue.Queue;
+import com.connexta.replication.api.queue.QueueBroker;
+import com.connexta.replication.api.queue.SiteQueue;
+import io.micrometer.core.instrument.MeterRegistry;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Stream;
+
+/**
+ * Provides a non-persistent implementation for the queue broker that keeps queues and tasks in
+ * memory.
+ */
+public class MemoryQueueBroker implements QueueBroker {
+  // TODO: need to monitor and poll site configurations for changes such that when sites are removed
+  // the queues are disposed. This might result in composite queues that no longer have queues in
+  // them
```

Comment:
Not sure I would want to have our micro services manage the queues when we are external. Don't forget that the query service is not the only services using these queues and managing them might end up being a lot more than simply removing an entry from a map (as it is right now in this in-memory implementation). 

This was meant to be in that spirit. 

Although, right now even if we leak those, I am not concern since this is mostly going to be used for testing going forward and for the current installment, we will only have 2 sites to deal with.